src/lib/server.ts

```diff
@@ -212,32 +215,56 @@ class Clients {
 }
 interface ZwavejsServerOptions {
   port: number;
+  logger?: Logger;
+}
+
+export interface Logger {
+  error(message: string | Error, error?: Error): void;
```

Comment:
I would say that all logger functions should define the argument lists as `...args: unknown[]`.

Reply:
I think typing it is ok, since that shows what we will pass to it. Typescript looks at the type, not the interface that defined the type, so if someone passes a compatible logging object, it's all good. With unknown we couldn't communicate to consumers of this lib if the data we pass changes in the future.

Reply:
While I agree on properly typing the interfaces whenever it's possible I don't see much sense to do it for logger's functions parameters, where we should be able to pass anything we want (as i.e. `console` family does it).

Reply:
I honestly don't _really_ care about the logger. For all I care we keep it undefined 🤷